packages/cheqd-sdk/src/dids/CheqdDidResolver.ts

```diff
@@ -0,0 +1,202 @@
+      await cheqdSdkLedgerService.connect({ network })
+
+      let resourceId: string
+      if (did.match(cheqdResourceRegex)?.input) {
+        resourceId = did.split('/')[2]
+      } else if (params && params.resourceName && params.resourceType) {
+        let resources = (await cheqdSdkLedgerService.resolveCollectionResources(id)).resources
+        resources = resources.filter(
+          (resource) => resource.name == params.resourceName && resource.resourceType == params.resourceType
+        )
+        if (!resources.length) {
+          throw new Error(`No resources found`)
+        }
+
+        let resource: Metadata | undefined
+        if (params.version) {
+          resource = resources.find((resource) => resource.version == params.version)
+        } else {
+          const date = params.time ? new Date(params.time) : new Date()
+          // find the resourceId to the created time
+          resources.sort(function (a, b) {
+            const distancea = Math.abs(date.getTime() - a.created!.getTime())
+            const distanceb = Math.abs(date.getTime() - b.created!.getTime())
+            return distancea - distanceb
+          })
+          resource = resources[0]
+        }
+
+        if (!resource) {
+          throw new Error(`No resources found`)
+        }
+
+        resourceId = resource.id
+      } else {
+        return {
+          error: 'notFound',
+          message: `resolver_error: Invalid did url '${did}'`,
+        }
+      }
+
+      const { resource, metadata } = await cheqdSdkLedgerService.resolveResource(id, resourceId)
+      if (!resource || !metadata) {
+        throw new Error('Please try again, Internal error')
+      }
+
+      const mimeType = metadata.mediaType
+      let result: any
+      if (mimeType == 'application/json') {
+        result = await JsonEncoder.fromBuffer(resource.data)
+      } else if (mimeType == 'text/plain') {
+        result = toString(resource.data)
+      } else {
+        result = toString(resource.data, 'base64')
+      }
+
+      return {
+        resource: result,
+        resourceMetadata: metadata,
+        resourceResolutionMetadata: {},
+      }
+    } catch (error) {
+      return {
+        error: 'notFound',
+        message: `resolver_error: Unable to resolve resource '${did}': ${error}`,
+      }
+    }
+  }
+
+  private async ResolveAllDidDocVersions(agentContext: AgentContext, parsedDid: ParsedCheqdDid) {
+    const cheqdSdkLedgerService = agentContext.dependencyManager.resolve(CheqdSdkLedgerService)
+    const { did, network } = parsedDid
+
+    await cheqdSdkLedgerService.connect({ network })
+    const { didDocumentVersionsMetadata } = await cheqdSdkLedgerService.resolveMetadata(did)
+    return {
+      didDocument: { id: did } as DidDocument,
+      didDocumentMetadata: didDocumentVersionsMetadata,
+      didResolutionMetadata: {},
+    }
+  }
+
+  private async DereferenceCollectionResources(agentContext: AgentContext, parsedDid: ParsedCheqdDid) {
```

Comment:
Curious: does this mean you can resolve did-linked resources using this did resolver?

Reply:
Yep. added a separate function resolveResource in the same class which can used to resolve the actual resource. A separate function was needed as the return type is different
Generally while resolving a didDocument the list of resource metadatas are added the didDocument metadata as well.